docs/pages/inboxes/disappearing-messages.md

```diff
@@ -0,0 +1,97 @@
+
+However, it's important to understand that:
+
+- A conversation participant using an app that doesn't support disappearing messages will not experience disappearing message behavior.
+- Messages aren't deleted from the XMTP network.
+
+Therefore, disappearing messages should be understood as best-effort, app-level privacy that helps avoid leaving an easily accessible record in a messaging UI. However, it's not a guaranteed, system-wide “message self-destruct.”
+
+## Implement disappearing messages
+
+Here is a high-level view of the disappearing messages flow:
+
+1. When a 1:1 or group chat conversation is created or updated, a user with appropriate permissions can set disappearing message conditions for messages sent in the conversation.
+2. In a conversation with disappearing message conditions, a sender creates a message. The message abides by disappearing message metadata set for the conversation.
+3. The message is sent over the XMTP network.
+4. The recipients’ apps with disappearing message support receive the message and check the expiration conditions set for the conversation.
+5. If the message hasn't expired, the app displays the message in the messaging UI.
+6. If the message has expired, the message is removed from (or not displayed in the first place in) the conversation participants’ app UIs and is deleted from the apps’ local storage.
+
+### Enable disappearing messages for a conversation
+
+When creating or updating a conversation, enable users with appropriate permissions to set disappearing message expiration conditions.
+
+This includes setting the following conditions expressed in nanoseconds (ns):
+
+- `from_ns`: Starting timestamp from which the message lifespan is calculated
+- `in_ns`: Duration of time during which the message should remain visible to conversation participants
+
+For example:
+
+1. Set `from_ns` to the current time, such as `1738620126404999936` (nanoseconds since the Unix epoch of January 1, 1970).
+2. Set `in_ns` to the message lifespan, such as 1800000000000000 (30 minutes).
+3. Use `from_ns` and `in_ns` to calculate the message expiration time of `1738620126404999936 + 1800000000000000 = 1740420126404999936`.
+
+To learn more see [conversation.rs](https://github.com/xmtp/libxmtp/blob/main/bindings_node/src/conversation.rs#L49).
+
+### Send a disappearing message
+
+When sending a message, it abides by message expiration conditions set for the conversation. For example:
+
+```text
+// Example: disappearing message with an expiration time based on a 
+// countdown start time and expiration countdown
+```
+
+### Receive a disappearing message
+
+On the receiving side, your app needs to:
+
+1. Check for disappearing message metadata set for the conversation. For example:
+    
+    ```text
+    // Example: check for disappearing message on the message's parent conversation
+    ```
+    
+2. Display the message if the current time is before the expiration time. For example:
```

Comment:
actually, we don't do that, receiving and processing a message happens in a separate flow, and we let the flow continue it's work as before, here the worker delete the messages that expired.